### DISCUSSION TITLE
Thread Panic and Row Deletion

### DISCUSSION THREAD

**Discussion by TankMan649** [lancedb/lancedb#1028] [Q&A]
I have an index with large amount of underlying rows and there turned out to be some empty rows in the tables. So when I run queries I get a "thread panicked" error. So how do I delete the empty data rows throughout the table across all empty rows. The total table comes from roughly 15 million records which are then chunked into section which probably results in at least 60 million rows. How do I delete the empty rows so I don't get the "thread panicked" error?

**Comment by wjones127** [comment]
Could you be more specific on what you mean by empty? Is there a text column that is null, or empty?

If it's null, you could do something like: `table.delete("my_column is null")`. If it's empty, could do something like `table.delete("my_column = ''")`.
